functions/src/modules/SQR.functions.ts

```diff
@@ -648,3 +648,33 @@ export const getSpareFiles = functions.https.onCall(
       .slice(0, count || 20);
   }
 );
+
+export const cancelAllotment = functions.https.onCall(
+    async (
+        { fileName, comments, token, status },
```

Comment:
Pass `reason` instead of the `status` here. This is function's logic to transform reason into a status change.